### ISSUE TITLE
/login /logout routes are generating 404 not found errors

### ISSUE THREAD

**Issue by bschooly** [auth0/express-openid-connect#359]
Not sure what has happened. I have a simple app that was working (login, logout worked), then something has changed. 

How do I debug this? I set DEBUG="express:*" and the output shows: 

  express:router use '/' <anonymous> +262ms
  express:router:layer new '/' +0ms
  express:router use '/' <anonymous> +1ms
  express:router:layer new '/' +0ms
  express:router:route new '/login' +1ms
  express:router:layer new '/login' +0ms
  express:router:route get '/login' +0ms
  express:router:layer new '/' +0ms
  express:router:route get '/login' +0ms
  express:router:layer new '/' +0ms
  express:router:route new '/logout' +0ms
  express:router:layer new '/logout' +0ms
  express:router:route get '/logout' +0ms
  express:router:layer new '/' +1ms
  express:router:route new '/callback' +0ms
  express:router:layer new '/callback' +0ms
  express:router:route get '/callback' +0ms
  express:router:layer new '/' +0ms
  express:router:route get '/callback' +0ms
  express:router:layer new '/' +0ms
  express:router:route get '/callback' +0ms
  express:router:layer new '/' +0ms
  express:router:route new '/callback' +0ms
  express:router:layer new '/callback' +0ms
  express:router:route post '/callback' +0ms
  express:router:layer new '/' +0ms
  express:router:route post '/callback' +0ms
  express:router:layer new '/' +0ms
  express:router:route post '/callback' +0ms
  express:router:layer new '/' +1ms
  express:router:route post '/callback' +0ms
  express:router:layer new '/' +0ms

Yet these routes don't work.

### Reproduction

- Step 1..
- Step 2..
- ...

### Environment

- **Version of this library used:**
- **Which framework are you using, if applicable:**
- **Other modules/plugins/libraries that might be involved:**
- **Any other relevant information you think would be useful:**

**Comment by bschooly** [comment]
Well, the error seems to be related to regex I'm using

**Comment by adamjmcgrath** [comment]
Hi @bschooly 

You need to escape the `/` delimiter, eg `/^(?!\/api).+/`

**Comment by adamjmcgrath** [comment]
Hi @bschooly,

> Frankly I struggle with the design choices.

Happy to accept feedback if you want to share which specific design choices you're struggling with

> I see that implicit flow is now out of favor,

Implicit with Form Post? I wasn't aware of this - where are you seeing that?

**Comment by bschooly** [comment]
OK, I stand corrected. From auth0.com: 

_Don't let the term "implicit" mislead you! Although OAuth now discourages the use of the implicit grant for obtaining access tokens in SPAs, the scenario addressed by Implicit Flow with Form Post is completely different and is unaffected by the security issues that led to discouraging use with SPAs. Specifically, Implicit Flow with Form Post applies to traditional web apps as opposed to SPAs. You obtain ID tokens as opposed to access tokens, which have a completely different intended use. The flow uses POST as opposed to placing tokens in URL fragments (as with SPAs) which can expose token bits to browser history attacks, redirect headers, and so on._

I guess I latched on the whole 'don't do implicit flow' because I was considering the possibility of SPAs in our ecosystem. Especially since you library can be used as "back end" authentication for a SPA. 

As for design choices, I'm trying to build a general purpose auth mechanism (eg. works for clients of our services whether they be traditional web apps, SPAs, or mobile) using oauth/openid.  Your library looks good for traditional web apps, possible for SPAs (though I'm uncertain on how to handle callbacks on reauth w/o loosing state (SPA reload)). As for Mobile, I haven't even looked yet. 

So my needs, I guess fall outside the scope of your library but if you have any pointers or suggestions on how to proceed, I'm all ears :)

**Comment by adamjmcgrath** [comment]
>  I'm trying to build a general purpose auth mechanism (eg. works for clients of our services whether they be traditional web apps, SPAs, or mobile)

Yep - outside the scope of this SDK. I suspect you want to combine more than one of our SDKs for this type of architecture.
